avalon/pipeline.py

```diff
@@ -720,6 +720,84 @@ def get_representation_context(representation):
     return context
 
 
+def update_current_context(context):
+    """Update active Session to new context.
+
+    The context can hold `asset`, `task` and `app`, other keys are ignored.
+
+    Args:
+        context (dict): The context to apply.
+
+    Returns:
+        dict: The changed key, values in the current Session.
+
+    """
+
+    mapping = {
+        "AVALON_ASSET": context.get("asset", None),
+        "AVALON_TASK": context.get("task", None),
+        "AVALON_APP": context.get("app", None),
+    }
+    changed = {key: value for key, value in mapping.items() if
```

Comment:
I would skip this.

Always update everything in the context. Regardless of whether no change was found. Down the line, when we're confident this works, we can look at making optimisations like these.